### PR TITLE
[TE] Clean up and inject ThirdEye Credentials into SecurityContext

### PR DESCRIPTION
**Pull request by akshayrai**

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java [hunk 1]
```diff
@@ -42,15 +51,23 @@ public void setPassword(String password) {
...
     if (this == o) {
       return true;
     }
-    if (!(o instanceof Credentials)) {
+    if (!(o instanceof ThirdEyeCredentials)) {
       return false;
     }
-    Credentials that = (Credentials) o;
+    ThirdEyeCredentials that = (ThirdEyeCredentials) o;
     return Objects.equals(principal, that.principal) && Objects.equals(password, that.password);
```
Comment (xiaohui-sun) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java]: Add token into comparison.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java [hunk 2]
```diff
@@ -22,9 +22,18 @@
 import java.util.Objects;
 
 
-public class Credentials {
+public class ThirdEyeCredentials {
   String principal;
   String password;
+  String token;
```
Comment (xiaohui-sun) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java]: Is it mainly for service account?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java (hunk 1 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java]: Added

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java (hunk 2 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeCredentials.java]: Token exists for all the users. This PR only addresses injecting the authenticated user into the `SecurityContext`. That way we can access the user from every api call.

I will send a followup PR to add the service restriction.

**Pull request merged**